libsolidity/interface/CompilerStack.cpp

```diff
@@ -459,7 +466,7 @@ void CompilerStack::link()
 
 vector<string> CompilerStack::contractNames() const
 {
-	if (m_stackState < AnalysisSuccessful)
+	if (m_stackState < AnalysisPerformed && !m_parserErrorRecovery)
```

Comment:
Same here.

Reply:
I believe we want contract names if parsing was not successful. I don't know of any other way to do this. If you do, please just go ahead and make the change.